packages/model-viewer-effects/src/effect-composer.ts

```diff
@@ -0,0 +1,321 @@
+/* @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactiveElement } from 'lit';
+import { EffectComposer, EffectPass, NormalPass, RenderPass, Selection, Pass } from 'postprocessing';
+import { disposeEffectPass, isConvolution } from './utilities.js';
+import { ModelViewerElement } from '@beilinson/model-viewer';
+import { $requireNormals, $requireSeparatePass, IMVEffect, MVEffectBase } from './effects/mixins/effect-base.js';
+import { ModelScene } from '@beilinson/model-viewer/lib/three-components/ModelScene.js';
+import { Camera, HalfFloatType, UnsignedByteType, WebGLRenderer } from 'three';
+import { property } from 'lit/decorators.js';
+import { OverrideMaterialManager } from 'postprocessing';
+
+OverrideMaterialManager.workaroundEnabled = true;
+
+export const $scene = Symbol('scene');
+
+export const $effectOptions = Symbol('effectOptions');
+export const $effectComposer = Symbol('effectComposer');
+export const $renderPass = Symbol('renderPass');
+export const $normalPass = Symbol('normalPass');
+export const $clearPass = Symbol('clearPass');
+export const $removeClearPass = Symbol('removeClearPass');
+export const $addClearPass = Symbol('addClearPass');
+export const $effectPasses = Symbol('effectsPass');
+export const $effects = Symbol('effects');
+export const $selection = Symbol('selection');
+export const $setSelection = Symbol('setSelection');
+export const $resetEffectPasses = Symbol('resetEffectPasses');
+export const $userEffectCount = Symbol('userEffectCount');
+
+/**
+ * Light wrapper around {@link EffectComposer} for storing the `scene` and `camera
+ * at a top level, and setting them for every {@link Pass} added.
+ */
+export class EffectRenderer extends EffectComposer {
```

Comment:
I'm super impressed with the work you've put in here - this is a powerful new set of capabilities. If someone wanted to extend this with new effects, but wanted to do it outside of our repo, what directions would you give them? Two cases: also depending on the `postprocessing` lib, or using something else.

Reply:
I think it would be practical to require the `postprocessing` library's EffectComposer, rather than the original three.js EffectComposer. Supporting additional effects that don't ship with `postprocessing` is worth considering, though.